OrbitService/UtilsTest.cpp

```diff
@@ -0,0 +1,163 @@
+    EXPECT_EQ(result.value().size(), 1);
+  }
+
+  {
+    // Example data
+    const std::string data{absl::StrFormat(
+        "7f6874285000-7f6874288000 r--p 00000000 fe:01 661216                     %s\n"
+        "7f6874288000-7f687428c000 r-xp 00003000 fe:01 661216                     %s\n"
+        "7f687428c000-7f687428e000 r--p 00007000 fe:01 661216                     %s\n"
+        "7f687428e000-7f687428f000 r--p 00008000 fe:01 661216                     %s\n"
+        "7f687428f000-7f6874290000 rw-p 00009000 fe:01 661216                     %s\n",
+        hello_world_path, hello_world_path, hello_world_path, hello_world_path, hello_world_path)};
+
+    const auto result = ListModules(data);
+    ASSERT_TRUE(result) << result.error().message();
+    ASSERT_EQ(result.value().size(), 1);
+    const orbit_grpc_protos::ModuleInfo& module_info = result.value()[0];
+    EXPECT_EQ(module_info.name(), "hello_world_elf");
+    EXPECT_EQ(module_info.file_path(), hello_world_path);
+    EXPECT_EQ(module_info.file_size(), 16616);
+    EXPECT_EQ(module_info.address_start(), 0x7f6874285000);
+    EXPECT_EQ(module_info.address_end(), 0x7f6874290000);
+    EXPECT_EQ(module_info.build_id(), "d12d54bc5b72ccce54a408bdeda65e2530740ac8");
+  }
+
+  // Is this actually going to work?
+}
+
+TEST(Utils, GetCpuUtilization) {
+  {
+    // Wrong data
+    const auto result =
+        GetCpuUtilization(std::string_view{"random example test data\nwith new line"});
+    EXPECT_FALSE(result) << result.error().message();
+  }
+
+  {
+    // Empty data
+    const auto result = GetCpuUtilization(std::string_view{""});
+    ASSERT_TRUE(result);
+    EXPECT_TRUE(result.value().empty());
+  }
+
+  {
+    // Currently running
+    const auto result = GetCpuUtilization();
+    EXPECT_TRUE(result) << result.error().message();
+  }
+  {
+    // Valid Example Data
+    std::string_view top_data{
+        "2636625,userA,20,0,4960340,271400,131228,R,118.8,0.4,7533:11,chrome\n"
+        "1,root,20,0,171340,12688,8456,S,0.0,0.0,6:37.00,systemd\n"
+        "2,root,20,0,0,0,0,S,0.0,0.0,0:00.61,kthreadd\n"
+        "3,root,0,-20,0,0,0,I,0.0,0.0,0:00.00,rcu_gp\n"};
+    const auto result = GetCpuUtilization(top_data);
+    ASSERT_TRUE(result);
+    EXPECT_EQ(result.value().size(), 4);
+
+    EXPECT_FLOAT_EQ(result.value().at(2636625), 118.8);
+    EXPECT_FLOAT_EQ(result.value().at(1), 0.0);
+    EXPECT_FLOAT_EQ(result.value().at(2), 0.0);
+    EXPECT_FLOAT_EQ(result.value().at(3), 0.0);
+  }
+}
+
+TEST(Utils, Is64Bit) {
+  {
+    const auto result = Is64Bit(getpid());
+    ASSERT_TRUE(result) << result.error().message();
+    EXPECT_TRUE(result.value());
+  }
+
+  {
+    const std::string data =
+        "/proc/1507305/exe: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically "
+        "linked, interpreter /lib64/ld-linux-x86-64.so.2, "
+        "BuildID[sha1]=16e95e7df322f1e9a8fb395dec25edc55e61fae7, for GNU/Linux 3.2.0, with "
+        "debug_info, not stripped";
```

Comment:
that test can go away if we use ElfFile.

Reply:
done